test cases/common/9 header install/meson.build

```diff
@@ -1,11 +1,12 @@
 project('header install')
 
-as_array = ['subdir.h']
+as_array = ['subdir.h', 'child/childdir.h', 'child/otherchild.h']
 
 subdir('vanishing_subdir')
 subdir('sub')
 
 h1 = install_headers('rootdir.h')
 h2 = install_headers(as_array, subdir : 'subdir')
-h3 = install_headers(subheader)
-h4 = install_headers(disabler())
+h3 = install_headers(as_array, subdir : 'subdir', preserve_path : true)
```

Comment:
I'm actually slightly surprised that this worked, since it seems you are installing `includedir/subdir.h` twice now (as that file doesn't have a leading directory component to preserve).

Reply:
I'm aware of that. I didn't think this was an issue, since meson doesn't complain about installing the same file twice. Unless this is destined to change, I don't see why to change that behaviour.

Reply:
Right, that was me commenting to myself.